src/helpers/calculationStepFunctions/numericCalculationFunctions.ts

```diff
@@ -0,0 +1,54 @@
+import { create, all } from 'mathjs'
+import { CurrentSurveyAnswer } from '@careevolution/mydatahelps-js'
+
+export const calculateValue = (calcString :string, surveyAnswers: CurrentSurveyAnswer[]) => {
+  const config = { }
+  const math = create(all, config)
+
+  let replacedString = replaceVariables(calcString, surveyAnswers);
+  replacedString = replacedString.replace("/{/g", "");
+  replacedString = replacedString.replace("/}/g", "");
+
+  if (replacedString.includes('Unmatched Variable')) {
+    return("Calculation Error - Unmatched Variable") 
+  }
+
+  return(math.evaluate(replacedString).toString());
+}
+
+function replaceVariables(expression: string, surveyAnswers: CurrentSurveyAnswer[]): string {
+  //find variables
+  const regex = /\{(.*?)\}/g;
+
+  //replace variables with numeric values and convert number to string
+  let replacedExpression = expression.toString().replace(regex, (match, variable) => {
+    const value = getValue(variable!, surveyAnswers);
+
+    if (typeof value === 'number' && !isNaN(value)) {
+      return value.toString();
+    }
+
+    return "Unmatched Variable";
+  });
+
+  return replacedExpression;
+}
+
+const getValue = (lookUpString :string, surveyAnswers :CurrentSurveyAnswer[]) => {
+  const splitString = lookUpString.split(",");
+  const stepIdentifier = splitString[0].trim()
+  const resultIdentifier = splitString.length > 1 ? splitString[1].trim() : undefined;
+
+  try {
+    return (resultIdentifier != undefined ? 
+        parseFloat(surveyAnswers.find((answer) => {
+            return(answer.stepIdentifier == stepIdentifier && answer.resultIdentifier == resultIdentifier);
+        })!.answers[0])
+        : parseFloat(surveyAnswers.find((answer) => {
+            return (answer.stepIdentifier == stepIdentifier);
+        })!.answers[0]));
```

Comment:
It's really interesting to see you tackle this because I've just recently been making a generic Calculation WebViewStepTemplate for another project.  It's interesting to compare/contrast how we went about this.  In my case, I knew *specifically* how the questions were going to be formatted (and that all of them were Form Steps), so I didn't try and make my version extremely generic.

Which brings me to my question here, wondering about constraints on how this step could actually be used.  Am I correct in that if the step was a Question Step with the Text Choice Answer Format, and a Text Choice Style of Multiple Choice that you could have multiple answers in the array here?  What's the expectation on how to calculate that type of step?  I could see an argument that it should be the sum of the parseFloat of the answers?

Reply:
Wonder what REDCap does for multiple choice.  If there is an easy way to match them maybe worth it.  Otherwise feels like we should ignore and not try to handle it

Reply:
<img width="727" alt="image" src="https://github.com/CareEvolution/MyDataHelpsUI/assets/60715340/add088ac-1505-4510-943f-717c7c8b15ef">

"If possible, avoid" haha. 

Maybe in our docs we say something similar, but explain the behavior if they do use it. Or should we say if length of the results for a step > 1 then throw an error.

Reply:
Yeah for an MVP the latter might make the most sense...just don't allow results > 1